pkg/internal/sqlsmith/scalar.go

```diff
@@ -373,6 +379,76 @@ func makeFunc(s *scope, ctx Context, typ types.T, refs colRefs) (tree.TypedExpr,
 	), typ), true
 }
 
+const maxWindowFrameOffset = 10
+
+func makeWindowFrame() *tree.WindowFrame {
+	// Window frame mode and start bound must always be present whereas end
+	// bound can be omitted.
+	frameMode := tree.WindowFrameMode(rand.Intn(3))
+	var startBound tree.WindowFrameBound
+	var endBound *tree.WindowFrameBound
+	if frameMode == tree.RANGE {
+		// RANGE mode is special in that if a bound is of type OffsetPreceding or
+		// OffsetFollowing, it requires that ORDER BY clause of the window function
+		// have exactly one column that can be only of the following types:
+		// DInt, DFloat, DDecimal, DInterval; so for now let's avoid this
+		// complication and not choose offset bound types.
+		// TODO(yuzefovich): fix this.
+		if coin() {
+			startBound.BoundType = tree.UnboundedPreceding
+		} else {
+			startBound.BoundType = tree.CurrentRow
+		}
+		if coin() {
+			endBound = new(tree.WindowFrameBound)
+			if coin() {
+				endBound.BoundType = tree.CurrentRow
+			} else {
+				endBound.BoundType = tree.UnboundedFollowing
+			}
+		}
+	} else {
+		// There are 5 bound types, but only 4 can be used for the start bound.
+		startBound.BoundType = tree.WindowFrameBoundType(rand.Intn(4))
+		if startBound.BoundType == tree.OffsetFollowing {
+			// With OffsetFollowing as the start bound, the end bound must be
+			// present and can either be OffsetFollowing or UnboundedFollowing.
+			endBound = new(tree.WindowFrameBound)
+			if coin() {
+				endBound.BoundType = tree.OffsetFollowing
+			} else {
+				endBound.BoundType = tree.UnboundedFollowing
+			}
+		}
+		if endBound == nil && coin() {
+			endBound = new(tree.WindowFrameBound)
+			// endBound cannot be "smaller" than startBound, so we will "prohibit" all
+			// such choices.
+			endBoundProhibitedChoices := int(startBound.BoundType)
+			if startBound.BoundType == tree.UnboundedPreceding {
+				// endBound cannot be UnboundedPreceding, so we always need to skip that
+				// choice.
+				endBoundProhibitedChoices = 1
+			}
+			endBound.BoundType = tree.WindowFrameBoundType(endBoundProhibitedChoices + rand.Intn(5-endBoundProhibitedChoices))
+		}
+		// We will set offsets regardless of the bound type, but they will only be
+		// used when a bound is either OffsetPreceding or OffsetFollowing. Both
+		// ROWS and GROUPS mode need non-negative integers as bounds.
+		startBound.OffsetExpr = tree.NewDInt(tree.DInt(rand.Intn(maxWindowFrameOffset)))
```

Comment:
So this is where sqlsmith gets cool. Instead of hard coding an integer, use `makeScalar(s, types.Int, refs)`, and take a `refs colRefs` argument to this func. It'll produce some funky thing that spits out on int based on the currently available column names, or make some constants.